### ISSUE TITLE
Re-route Keyboard Commands to Generator Actions

### ISSUE THREAD

**Issue by JeffreyGaydos** [TwinedRope/TwinedRope.github.io#19]
- The following shortcuts could be used
  - ✅ Ctrl+S: Save Project (unless this is blocked by the save webpage thing)
  - ✅ Ctrl+Z: Undo
  - ✅ Ctrl+Y: Redo
  - 🆕☑️ Ctrl+X: Cut (unimplemented button [but also just copy and delete recursive])
    - Partially done, the icon and actually implementing this will be done later (It has the variables, but no logic)
  - ✅ Ctrl+V: Paste (unlinked)
  - ✅ Ctrl+Shift+V: Paste (linked)
  - ✅ Ctrl+C: Copy
  - ✅ delete: Delete Recursive
- Additionally, support for navigating dialogue via arrow key could be cool
  - ~Make up arrow move up (ignoring collapsed children) (if exists)~
  - ~Make down arrow move down (ignoring collapsed children) (if exists)~
    - Removed due to pre-programmed scrolling effect, and I don't think holding ALT will be very ergonomic
  - ✅ Make left arrow move to the parent (if exists)
  - ✅ Make right arrow move to the child (if exists), expanding if necessary

**Comment by JeffreyGaydos** [comment]
We added some more shortcuts and modified others:
Save: now Ctrl+Alt+S to avoid the common save webpage browser shrotcut
✅ Toggle Sequence Numbers: Ctrl+Alt+T
✅ Import Project: Ctrl+I

Arrows are unimeplemented and likely will be unable to be without a held Alt or something else...

✅ The only issue is that once you open default browser shortcut, anything you were holding gets "stuck" on, so just pressing 1 key will result in the action. It would appear that you cannt easily override these, otherwise I would. Regardless, this causes the necessity to reset ctrl and shift immediately after every action, but this prevents repeated actions from working (i.e. Ctrl+V+V+V+V+V+...)

**Comment by JeffreyGaydos** [comment]
✅ When focused on the #input box, you need to cancel all possible actions, and allow the input specific actions (Shift+Enter)
- This still has the issue of adding an enter at the end of the input
- Consider adding the option to auto submit instead
  - ✅ This has been rolled into a separate issue

**Comment by JeffreyGaydos** [comment]
Implemented using a timeout, which gives me some serious inspiration to start using asynchronous calls for the rendering of things
